app/models/friendship.rb

```diff
@@ -1,16 +1,18 @@
 class Friendship < ApplicationRecord
   belongs_to :user
   belongs_to :friend, class_name: 'User'
-
   before_create :set_status
+
   # query  to check if friend ship all ready exits
   scope :friend_requests_sent, -> {where("status = ? ", 'Pending')}
+
   scope :confirmed_friends,
         -> (user) {where('user_id = ? AND status  IN (?) OR friend_id = ? AND status  IN (?) ' ,
-                                          user,['Confirmed','Pending'],user, ['Confirmed', 'Pending'])}
+                                          user,['Confirmed'],user, ['Confirmed'])}
   scope :awaiting_response, -> (user) {where("friend_id = ?  AND status = ?",user, 'Pending')}
-  scope :friendship_exist?, -> (user,friend) {where('user_id = ? AND status = ? AND friend_id = ? AND status = ? ' ,
-                                                                               user,'Pending',friend, 'Pending')}
+
+  scope :friendship_exist?, -> (user,friend) {where('user_id = ? AND status  IN (?) OR friend_id = ? AND status  IN (?) ',
```

Comment:
We can have a `before_validation` callback that checks whether a friendship/friend request already exists between the two users in question. This will simplify our logic for the controllers and views.

Reply:
> We can have a `before_validation` callback that checks whether a friendship/friend request already exists between the two users in question. This will simplify our logic for the controllers and views.

Thanks for your thorough review :100:  Will start implementing the required changes :+1: 

Just to address some concerns.

- [ ] The plan is to implement React for this project so that's why all the javascript stuff is there. It was just too much extra work to do right now lol.

- [ ] Too add a friend the little icon next to the person's name is really a button.

Nice catch with `user.friends`  will address this and the others changes thanks again.